pkg/admission/util.go

```diff
@@ -29,42 +29,86 @@ import (
 	"github.com/apache/yunikorn-k8shim/pkg/log"
 )
 
-func updatePodLabel(pod *v1.Pod, namespace string, generateUniqueAppIds bool, defaultQueueName string) map[string]string {
-	existingLabels := pod.Labels
-	result := make(map[string]string)
-	for k, v := range existingLabels {
-		result[k] = v
+func getNewApplicationInfo(pod *v1.Pod, namespace string, generateUniqueAppIds bool, defaultQueueName string) (map[string]string, map[string]string) {
+	newLabels := make(map[string]string)
+	newAnnotations := make(map[string]string)
+
+	appID, isAppIdFromLabel := getApplicationIDFromPod(pod)
+	disableStateAware, isDisableStateAwareFromLabel := getDisableStateAwareFromPod(pod)
+
+	// for undefined configuration, am_conf will add 'root.default' as default queue name
+	// if a custom name is configured for default queue, it will be used instead of root.default
+	queueName, isQueueNameFromLabel := utils.GetQueueNameFromPod(pod, defaultQueueName)
+
+	if isAppIdFromLabel || isDisableStateAwareFromLabel || isQueueNameFromLabel {
+		log.Log(log.Admission).Warn("Pod contains label for Yunikorn configuration. This is deprecated and will be ignored in a future release. Please migrate to annotation-based configuration.",
```

Comment:
I've already talked to @craigcondit about this, we might not be in a position to ever remove this, it's going to be too dangerous. Also, certain settings actually should be specified as labels because that allows pods to be grouped/selected based on certain criteria. Let's think about it.

Reply:
I agree with the benefits of using label in terms of grouping and selecting.
Deprecate it might result in users unable to implement alternative solution.

Regarding this PR:
1. always prefer annotation above label.
2. Move labels to annotation, but keep existing labels before deprecation.

Before making any change, I think we need to align the scope again.
Below are my questions if we are not going to deprecate the labels:
1. Will AM patch missing attributes to label only?  Or patch to label and annotation both?
-> I think for  'applicationID' and 'queue', we should patch to both.
2. How about the existing label "disableStateAware"?
-> I think we can just change it to annotation because it's only be set in AM.  
-> This label could be deprecated. (Or suggest not using it)
3. If we have a different applicationID set in label/annotation,  do we still prefer annotation above label.
-> I think the answer is yes for consistency, we must guide user not to set different value if they require labels for selecting/grouping.
4. If only annotations were set in pod, should AM patch it(app-id, queue) to label?
-> Yes, align with my answer of question 1
5. If only labels were set in pod, should AM patch it(applicationId, queue) to annoataion?
-> Yes, align with my answer of question 1

My summary of new version:
1. Always prefer annotation above labels
2. AM ensure applicationID/queue existing in label and annotation.
3. Change label "disableStateAware" to annotation.

@pegasas, do you have any feedback?

Reply:
Also needs @wilfred-s 's review too.

Reply:
Sorry, I just pinged the wrong person.

@pbacsko,@craigcondit , do you have any feedback for the new version AM behaviror?

Reply:
@pbacsko, @craigcondit ,
After reconsideration, if deprecated labels are no longer needed, I can revert most of the AM changes to keep this PR simple. We can maintain the previous UpdateLabel behavior for AM. Just make sure to always get annotation before labels when accessing the value.

Will update a version with this idea, please let me know if you have any concern.

Reply:
I'd like to get @wilfred-s to weigh in on this one.